src/main/kotlin/eu/gillissen/rider/usersecrets/OpenUserSecretsAction.kt

```diff
@@ -2,75 +2,107 @@ package eu.gillissen.rider.usersecrets
 
 import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
 import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.fileEditor.FileEditorManager
-import com.intellij.openapi.project.DefaultProjectFactory
-import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.LocalFileSystem
 import com.jediterm.terminal.ui.UIUtil
+import com.jetbrains.rider.run.environment.MSBuildEvaluator
 import java.io.File
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.Task
+import java.util.concurrent.TimeUnit
 import javax.xml.parsers.DocumentBuilderFactory
 
 class OpenUserSecretsAction : AnAction() {
 
-    private var chosenProject: Project = DefaultProjectFactory.getInstance().defaultProject
-    private var id: String = ""
-
-    override fun actionPerformed(actionEvent: AnActionEvent) {
-        val dirs = getSecretsDirectory()
-        val path = "$dirs${File.separatorChar}secrets.json"
-        val file = File(path)
-        if (!file.exists()) {
-            File(dirs).mkdirs()
-            file.createNewFile()
-            file.writeText("{\n" +
-                    "//    \"MySecret\": \"ValueOfMySecret\"\n" +
-                    "}")
-        }
-        val virtualFile = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(file)
-            ?: return
-        FileEditorManager.getInstance(chosenProject).openFile(virtualFile, true)
-    }
-
-    private fun getSecretsDirectory(): String {
-        return if (UIUtil.isWindows)
-            "${System.getenv("APPDATA")}${File.separatorChar}microsoft${File.separatorChar}UserSecrets${File.separatorChar}$id"
-        else
-            "${System.getenv("HOME")}${File.separatorChar}.microsoft${File.separatorChar}usersecrets${File.separatorChar}$id"
+    companion object {
+        private const val CsProjExtension = "csproj"
+        private const val UserSecretsIdMsBuildProperty = "UserSecretsId"
     }
 
     override fun update(actionEvent: AnActionEvent) {
-        val project = actionEvent.getData(PlatformDataKeys.PROJECT)
 
-        if (project == null || project.isDefault) {
-            actionEvent.presentation.isVisible = false
-            actionEvent.presentation.isEnabled = false
+        val project = CommonDataKeys.PROJECT.getData(actionEvent.dataContext) ?: return
+
+        if (project.isDefault) {
+            actionEvent.presentation.isEnabledAndVisible = false
             return
         }
```

Comment:
This deviates from the original behaviour where `isEnabledAndVisible` is set to `false` when `project == null`.

In the case that it's not necessary to set it to false, please update the other branches as well.

Reply:
Good catch, updating.